src/input_common/joycon/jc_adapter.cpp

```diff
@@ -0,0 +1,372 @@
+// Copyright 2014 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include <chrono>
+#include <thread>
+#include "common/logging/log.h"
+#include "input_common/joycon/Jc_adapter.h"
+
+namespace JCAdapter {
+
+/// Used to loop through and assign button in poller
+constexpr std::array<PadButton, 12> PadButtonArray{
+    PadButton::PAD_BUTTON_LEFT, PadButton::PAD_BUTTON_RIGHT, PadButton::PAD_BUTTON_DOWN,
+    PadButton::PAD_BUTTON_UP,   PadButton::PAD_TRIGGER_Z,    PadButton::PAD_TRIGGER_R,
+    PadButton::PAD_TRIGGER_L,   PadButton::PAD_BUTTON_A,     PadButton::PAD_BUTTON_B,
+    PadButton::PAD_BUTTON_X,    PadButton::PAD_BUTTON_Y,     PadButton::PAD_BUTTON_START,
+};
+
+Joycons::Joycons() {
+    if (joycon_handle != nullptr) {
+        return;
+    }
+    LOG_INFO(Input, "JC Adapter Initialization started");
+
+    const int init_res = hid_init();
+    if (init_res ==0) {
+        Setup();
+    } else {
+        LOG_ERROR(Input, "HIDapi could not be initialized. failed with error = {}", init_res);
+    }
+}
+
+void Joycons::Set_rumble(f32 amp_high, f32 amp_low, f32 freq_high, f32 freq_low) {
+    hdRumble.freq_high = std::clamp(freq_high, 81.75177f, 1252.572266f);
+    hdRumble.freq_low = std::clamp(freq_low, 40.875885f, 626.286133f);
+    hdRumble.amp_high = std::clamp(amp_high, 0.0f, 1.0f);
+    hdRumble.amp_low = std::clamp(amp_low, 0.0f, 1.0f);
+    SendRumble();
+}
+void Joycons::SendRumble() {
+    u8* buffer = new u8[49];
```

Comment:
```suggestion
    std::array<u8, 49> buffer{};
```

Will allow you to also remove the loop below this.